### PR TITLE
Deploy docs from GitHub Actions

### PR DESCRIPTION
**Pull request by banesullivan**
Work to automate the documentation deployment for #5344 

This re-enables deployment of our development docs from the main branch and will generate docs previews for all PRs.

Work to deploy release docs will come later

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -13,6 +13,9 @@
 
 Overview
 ********
+
+This is a test message to see if the deployment is working!
+
```
Comment (banesullivan) [inline doc/source/index.rst:18]: Simply to test if the built docs are indeed being deployed. Will remove.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -79,6 +79,13 @@ jobs:
           path: ${{ env.PYVISTA_EXAMPLE_DATA_PATH }}
           key: example-data-1-${{ hashFiles('pyvista/_version.py') }}
 
+      - name: Cache Sphinx build directory
+        uses: actions/cache@v3
+        if: env.USE_CACHE == 'true' && !startsWith(github.ref, 'refs/heads/release/') && !startsWith(github.ref, 'refs/tags/v')
+        with:
+          path: doc/_build/
+          key: doc-examples-${{ hashFiles('pyvista/_version.py') }}
+
```
Comment (banesullivan) [inline .github/workflows/docs.yml:88]: I want to reflect on whether or not we actually want to do this

File: doc/source/index.rst (hunk 1 above)
Comment (banesullivan) [reply doc/source/index.rst:18]:
```suggestion
```

**Comment by banesullivan** [comment]
@tkoyama010, let's go ahead and merge this to get the docs re-deploying from the main branch. This will also create a preview from the already generated docs build for every PR. I will have to follow up sometime between now and our next release with an automation for release docs

**Review by tkoyama010 — APPROVED** [review]
> @tkoyama010, let's go ahead and merge this to get the docs re-deploying from the main branch. This will also create a preview from the already generated docs build for every PR. I will have to follow up sometime between now and our next release with an automation for release docs

Roger. Thanks for working this 🙏

**Comment by user27182** [comment]
> This re-enables deployment of our development docs from the main branch and will generate docs previews for all PRs.

Is the intention for documentation previews to also work with PRs from forks? If so, this does not seem to be working at the moment e.g.
https://github.com/pyvista/pyvista/actions/runs/9900544390/job/27353216930?pr=6353#step:3:17

> Netlify credentials not provided, not deployable

(Also, the action shows a green checkmark, suggesting there was no issue even though it's not working)

To get the previews to work, should I create new PRs from `pyvista` branches directly instead of from a fork?

**Comment by user27182** [comment]
> (Also, the action shows a green checkmark, suggesting there was no issue even though it's not working)

Seems like the option controls this (from https://github.com/nwtgck/actions-netlify):

`fails-without-credentials` Fails if no credentials provided (default: false)

**Comment by banesullivan** [comment]
>for documentation previews to also work with PRs from forks?

Unfortunately, no these will not work from forks since the GitHub Actions secrets are not available under forks. I didn't realize this would consequently cause a CI failure for the docs build from any fork PR. 

> Seems like the option controls this (from https://github.com/nwtgck/actions-netlify):
> fails-without-credentials Fails if no credentials provided (default: false)

This seems like the right thing for us to specify to make sure forked PRs do not fail

**Comment by banesullivan** [comment]
> To get the previews to work, should I create new PRs from pyvista branches directly instead of from a fork?

Correct, we can only generate previews if the PR was made from a branch under the pyvista repo which you are more than welcome to create

**Comment by tkoyama010** [comment]
> > To get the previews to work, should I create new PRs from pyvista branches directly instead of from a fork?
> 
> Correct, we can only generate previews if the PR was made from a branch under the pyvista repo which you are more than welcome to create

Sorry. I had missed that change. I had made it so that any branch can be previewed if the administrator comments on the command, so that we can preview when the first contributor sends a pull request. Was it intentional to lose that?

**Comment by banesullivan** [comment]
> if the administrator comments on the command, so that we can preview when the first contributor sends a pull request. Was it intentional to lose that?

I did not know that was possible! Let me think about whether we want to enable this and look at the previous implementation. My main intention was to prevent there from being multiple CI workflows that would build the docs

**Comment by tkoyama010** [comment]
> > if the administrator comments on the command, so that we can preview when the first contributor sends a pull request. Was it intentional to lose that?
> 
> I did not know that was possible! Let me think about whether we want to enable this and look at the previous implementation. My main intention was to prevent there from being multiple CI workflows that would build the docs

I added it in #6395.

**Pull request merged**